source/opt/propagator.h

```diff
@@ -36,7 +36,7 @@ struct Edge {
   ir::BasicBlock* source;
   ir::BasicBlock* dest;
   bool operator<(const Edge& o) const {
-    return source < o.source || dest < o.dest;
+    return source->id() + dest->id() < o.source->id() + o.dest->id();
```

Comment:
At first I worried about unsigned overlflow. But that has defined behaviour.  Also, SPIR-V universal limits bound IDs to at most 0x3ffff.

Reply:
Yeah, Given the limits on instructions, I figured the id space cannot be bigger than 16 bits.

Reply:
I don't think edges (2, 4) and (1, 5) should be considered equal. Please use std::pair, std::tie or manual comparisons:
```
if (source < o.source)
  return true;
if (o.source < source)
  return false;
if (dest < o.dest)
  return true;
if (o.dest < dest)
  return false;
```